lib/cli.js

```diff
@@ -17,86 +17,82 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var getGitFolder = require('./getGitFolder');
 var validateMessage = require('../index');
 
-// hacky start if not run by mocha :-D
-// istanbul ignore next
-if (process.argv.join('').indexOf('mocha') === -1) {
-  var bufferToString = function (buffer) {
-    var hasToString = buffer && typeof buffer.toString === 'function';
-    return hasToString && buffer.toString();
-  };
+var commitMsgFileOrText = process.argv[2];
+var commitErrorLogPath = process.argv[3];
 
-  var getFileContent = function (filePath) {
-    try {
-      var buffer = fs.readFileSync(filePath);
-      return bufferToString(buffer);
-    } catch (err) {
-      // Ignore these error types because is most likely it is validating
-      // a commit from a text instead of a file
-      if(err && err.code !== 'ENOENT' && err.code !== 'ENAMETOOLONG') {
-        throw err;
-      }
-    }
-  };
+// On running the validation over a text instead of git files such as COMMIT_EDITMSG and GITGUI_EDITMSG
+// is possible to be doing that the from anywhere. Therefore the git directory might not be available.
+var gitDirectory;
+try {
+  gitDirectory = getGitFolder();
+
+  if (!commitErrorLogPath) {
+    commitErrorLogPath = path.resolve(gitDirectory, 'logs/incorrect-commit-msgs');
+  }
+} catch (err) {}
+
+var bufferToString = function (buffer) {
+  var hasToString = buffer && typeof buffer.toString === 'function';
+
+  return hasToString && buffer.toString();
+}
+
+var getCommit = function (msgFileOrText) {
+  var commit = getCommitFromFile(msgFileOrText);
+
+  if (commit === null && !msgFileOrText) {
```

Comment:
Hmm, didn't see that comment since the change was made and pushed before I checked the PR again.

So there was a bug in the code I didn't see. Hence why the cleverness of your original design was flawed for maintainability.

Instead of bickering over why you feel things are inconsistent and attempting to use my own words back at me, accept that proper tooling isn't in place for consistency checks across the whole system. The previous maintainer had his way of wanting code styled and didn't have the tooling setup to enforce it. Now, I'm the lead maintainer and I'd like us to move to a more declarative structure. It's that simple and nothing to be upset over.